votingsystem/src/general_meeting.rs

```diff
@@ -0,0 +1,248 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::common::*;
+pub use ckey::Ed25519Public as Public;
+use coordinator::context::SubStorageAccess;
+use coordinator::module::*;
+use coordinator::types::*;
+pub use ctypes::BlockId;
+use foundry_module_rt::UserModule;
+use parking_lot::RwLock;
+use rand::Rng;
+use remote_trait_object::raw_exchange::{import_service_from_handle, HandleToExchange, Skeleton};
+use remote_trait_object::{Context as RtoContext, Service, ServiceRef};
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+enum ExecuteError {
+    InvalidMetadata,
+    InvalidSign,
+    InvalidFormat,
+    InvalidAdmin,
+}
+
+struct Context {
+    pub storage: Option<Box<dyn SubStorageAccess>>,
+}
+
+impl Context {
+    fn storage(&self) -> &dyn SubStorageAccess {
+        self.storage.as_ref().unwrap().as_ref()
+    }
+
+    fn storage_mut(&mut self) -> &mut dyn SubStorageAccess {
+        self.storage.as_mut().unwrap().as_mut()
+    }
+}
+
+impl Service for Context {}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TxGeneralMeeting {
+    pub number_of_agendas: u32,
+    pub end_time: TimeStamp,
+    pub tallying_time: TimeStamp,
+}
+
+impl Action for TxGeneralMeeting {}
+pub type OwnTransaction = crate::common::SignedTransaction<TxGeneralMeeting>;
+
+impl Context {
+    fn excute_tx(&mut self, transaction: &Transaction) -> Result<GeneralMeetingId, ExecuteError> {
+        if transaction.tx_type() != "CreateGeneralMeeting" {
+            return Err(ExecuteError::InvalidMetadata)
+        }
+        let tx: OwnTransaction =
+            serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+        tx.verify().map_err(|_| ExecuteError::InvalidSign)?;
+
+        let num_agandas = tx.tx.action.number_of_agendas;
+        let end_time = tx.tx.action.end_time;
+        let tallying = tx.tx.action.tallying_time;
+        let meeting = GeneralMeeting::new(end_time, tallying, num_agandas);
+        let key = meeting.id.as_ref();
+
+        self.storage_mut().set(key, serde_cbor::to_vec(&meeting).unwrap());
+        Ok(meeting.id)
+    }
+}
+
+impl Stateful for Context {
+    fn set_storage(&mut self, storage: ServiceRef<dyn SubStorageAccess>) {
+        self.storage.replace(storage.unwrap_import().into_remote());
+    }
+}
+
+impl InitGenesis for Context {
+    fn begin_genesis(&mut self) {}
+
+    fn init_genesis(&mut self, config: &[u8]) {
+        let admin: Public = serde_cbor::from_slice(&config).unwrap();
+        self.storage_mut().set(admin.as_ref(), serde_cbor::to_vec(&admin).unwrap());
```

Comment:
Why did you use the admin's public key as the "key"?
How will use read the value?